In [2]:
import os 
print(os.getcwd())

C:\Users\gabri\Downloads


In [13]:
import pandas as pd
import numpy as np
import os
#Q1:Defina um diretório de trabalho 
f='C:/Users/gabri/OneDrive'
os.chdir(f)
print(os.getcwd())

C:\Users\gabri\OneDrive


In [14]:
#Q2:Importe os 3 arquivos de dados para o Python

#arquivo 1
A='https://gitlab.com/lemaufpb/dspy/-/raw/main/data/dim_escola.xlsx'
d_escolas=pd.read_excel(A,sheet_name='dados',
                        names=['cod_escola','cod_municipio','cod_rede','localizacao','nome_escola','situacao_funcionamento'])
d_escolas

cod_escola  cod_municipio  cod_rede  localizacao  \
0       25000012        2501005         3            1   
1       25000020        2502003         3            1   
2       25000047        2502003         2            1   
3       25000055        2502003         2            1   
4       25000063        2502003         3            1   
...          ...            ...       ...          ...   
4611    25291920        2505238         3            1   
4612    25292927        2505238         3            2   
4613    25302949        2514800         3            2   
4614    25307924        2501575         3            1   
4615    25309935        2506400         3            1   

                                            nome_escola  \
0                                       EMEF JOAO ALVES   
1            CRECHE INTEGRADA MARIA DO SOCORRO LINHARES   
2                        ECI NOSSA SENHORA DA CONCEICAO   
3                                    EEEFM AMERICO MAIA   
4                      EMEIF PREFEITO PLINIO FORTE MAIA   
...                                                 ...   
4611        CRECHE INFANTIL NEMEZIO AUGUSTO DE MEIRELES   
4612                       EMEF JOAO CALIXTO DOS SANTOS   
4613              ESCOLA MUNICIPAL ANA CELIA DE QUEIROZ   
4614  CEMEB PROFESSORA MARLY BARBOSA DE ALMEIDA - TI...   
4615      EMEIF PROFESSORA CARMEM SILVIA ALVES DA SILVA   

      situacao_funcionamento  
0                          1  
1                          2  
2                          1  
3                          1  
4                          1  
...                      ...  
4611                       1  
4612                       1  
4613                       2  
4614                       1  
4615                       1  

[4616 rows x 6 columns]

In [15]:
#arquivo 2
B='https://gitlab.com/lemaufpb/dspy/-/raw/main/data/fato_rendimento_escolar.csv'
rend_escolar=pd.read_csv(B, sep='|',skiprows=5, na_values='-').\
rename(columns={'Ano':'Anos','Código da Escola':'cod_escola',
                             'Taxa de Abandono no Ensino Médio (%)':'Taxa_de_Abandono'})
rend_escolar['Taxa_de_Abandono'] = rend_escolar['Taxa_de_Abandono'].str.replace(',','.').astype(float)
rend_escolar

Anos  cod_escola  Taxa_de_Abandono
0     2014    25000047               NaN
1     2015    25000047              13.2
2     2016    25000047              12.7
3     2017    25000047               5.5
4     2018    25000047              10.8
...    ...         ...               ...
3514  2017    25284940               2.1
3515  2018    25284940               1.4
3516  2019    25284940               0.0
3517  2020    25284940               0.0
3518  2021    25284940               0.3

[3519 rows x 3 columns]

In [16]:
#arquivo 3
l='C:/Users/gabri/OneDrive/prova2/RELATORIO_DTB_BRASIL_MUNICIPIO (1).xls'

colunas=['UF','Nome_UF','cod_Região_Intermediária','Nome_Região_Intermediária',
         'cod_Região_Imediata','Nome_Região_Imediata','cod_Mesorregião',
         'Nome_Mesorregião','cod_Microrregião','Nome_Microrregião',
         'Município','cod_municipio','Nome_Município' ]
rdm=pd.read_excel(l,names=colunas)
rdm['id'] =rdm.index
rdm

UF           Nome_UF  cod_Região_Intermediária  \
0     11          Rondônia                      1102   
1     11          Rondônia                      1102   
2     11          Rondônia                      1101   
3     11          Rondônia                      1102   
4     11          Rondônia                      1101   
...   ..               ...                       ...   
5565  52             Goiás                      5201   
5566  52             Goiás                      5202   
5567  52             Goiás                      5206   
5568  52             Goiás                      5205   
5569  53  Distrito Federal                      5301   

             Nome_Região_Intermediária  cod_Região_Imediata  \
0                            Ji-Paraná               110005   
1                            Ji-Paraná               110005   
2                          Porto Velho               110002   
3                            Ji-Paraná               110004   
4                          Porto Velho               110002   
...                                ...                  ...   
5565                           Goiânia               520002   
5566                         Itumbiara               520009   
5567  Luziânia - Águas Lindas de Goiás               520022   
5568                Porangatu - Uruaçu               520018   
5569                  Distrito Federal               530001   

            Nome_Região_Imediata  cod_Mesorregião   Nome_Mesorregião  \
0                         Cacoal                2  Leste Rondoniense   
1                         Cacoal                2  Leste Rondoniense   
2                      Ariquemes                2  Leste Rondoniense   
3                      Ji-Paraná                2  Leste Rondoniense   
4                      Ariquemes                2  Leste Rondoniense   
...                          ...              ...                ...   
5565                    Anápolis                5         Sul Goiano   
5566                 Piracanjuba                5         Sul Goiano   
5567             Flores de Goiás                4       Leste Goiano   
5568  Ceres - Rialma - Goianésia                4       Leste Goiano   
5569            Distrito Federal                1   Distrito Federal   

      cod_Microrregião    Nome_Microrregião  Município  cod_municipio  \
0                    6               Cacoal         15        1100015   
1                    6               Cacoal        379        1100379   
2                    3            Ariquemes        403        1100403   
3                    5     Alvorada D'Oeste        346        1100346   
4                    3            Ariquemes         23        1100023   
...                ...                  ...        ...            ...   
5565                16         Pires do Rio      22005        5222005   
5566                15           Meia Ponte      22054        5222054   
5567                12  Entorno de Brasília      22203        5222203   
5568                12  Entorno de Brasília      22302        5222302   
5569                 1             Brasília        108        5300108   

               Nome_Município    id  
0       Alta Floresta D'Oeste     0  
1     Alto Alegre dos Parecis     1  
2                Alto Paraíso     2  
3            Alvorada D'Oeste     3  
4                   Ariquemes     4  
...                       ...   ...  
5565               Vianópolis  5565  
5566           Vicentinópolis  5566  
5567                 Vila Boa  5567  
5568            Vila Propício  5568  
5569                 Brasília  5569  

[5570 rows x 14 columns]

In [17]:
# Ao final faça uma inspeção nos dados usando .info() em cada objeto.

d_escolas.info()

rend_escolar.info()

rdm.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4616 entries, 0 to 4615
Data columns (total 6 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   cod_escola              4616 non-null   int64 
 1   cod_municipio           4616 non-null   int64 
 2   cod_rede                4616 non-null   int64 
 3   localizacao             4616 non-null   int64 
 4   nome_escola             4616 non-null   object
 5   situacao_funcionamento  4616 non-null   int64 
dtypes: int64(5), object(1)
memory usage: 216.5+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3519 entries, 0 to 3518
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Anos              3519 non-null   int64  
 1   cod_escola        3519 non-null   int64  
 2   Taxa_de_Abandono  3115 non-null   float64
dtypes: float64(1), int64(2)
memory usage: 82.6 KB
<class 'pandas.core.frame.DataFram

In [18]:
#Q3:Crie um objeto chamado df a partir da junção das tabelas dim_escola e dtb e dos seguintes filtros:
#escolas da rede estadual de ensino; em atividade; no estado da Paraíba.

p=pd.merge(rdm,d_escolas , on='cod_municipio', how='left')
df=p.query('Nome_UF=="Paraíba"').\
query('cod_rede==2').\
query('situacao_funcionamento == 1')                                                        
df

UF  Nome_UF  cod_Região_Intermediária Nome_Região_Intermediária  \
1243  25  Paraíba                      2503                     Patos   
1267  25  Paraíba                      2503                     Patos   
1284  25  Paraíba                      2502            Campina Grande   
1286  25  Paraíba                      2502            Campina Grande   
1291  25  Paraíba                      2502            Campina Grande   
...   ..      ...                       ...                       ...   
5828  25  Paraíba                      2503                     Patos   
5840  25  Paraíba                      2504        Sousa - Cajazeiras   
5845  25  Paraíba                      2504        Sousa - Cajazeiras   
5846  25  Paraíba                      2503                     Patos   
5855  25  Paraíba                      2502            Campina Grande   

      cod_Região_Imediata Nome_Região_Imediata  cod_Mesorregião  \
1243               250009                Patos                1   
1267               250010           Itaporanga                1   
1284               250005       Campina Grande                3   
1286               250005       Campina Grande                3   
1291               250005       Campina Grande                3   
...                   ...                  ...              ...   
5828               250009                Patos                2   
5840               250014                Sousa                1   
5845               250014                Sousa                1   
5846               250009                Patos                1   
5855               250007             Monteiro                2   

       Nome_Mesorregião  cod_Microrregião           Nome_Microrregião  \
1243   Sertão Paraibano                 7           Serra do Teixeira   
1267   Sertão Paraibano                 5                      Piancó   
1284  Agreste Paraibano                15             Brejo Paraibano   
1286  Agreste Paraibano                15             Brejo Paraibano   
1291  Agreste Paraibano                15             Brejo Paraibano   
...                 ...               ...                         ...   
5828          Borborema                 8  Seridó Ocidental Paraibano   
5840   Sertão Paraibano                 3                       Sousa   
5845   Sertão Paraibano                 3                       Sousa   
5846   Sertão Paraibano                 3                       Sousa   
5855          Borborema                10            Cariri Ocidental   

      Município  cod_municipio Nome_Município    id  cod_escola  cod_rede  \
1243        106        2500106    Água Branca  1242  25033204.0       2.0   
1267        205        2500205         Aguiar  1243  25023489.0       2.0   
1284        304        2500304  Alagoa Grande  1244  25062638.0       2.0   
1286        304        2500304  Alagoa Grande  1244  25062670.0       2.0   
1291        304        2500304  Alagoa Grande  1244  25062816.0       2.0   
...         ...            ...            ...   ...         ...       ...   
5828      17100        2517100         Várzea  1461  25040600.0       2.0   
5840      17209        2517209    Vieirópolis  1462  25019740.0       2.0   
5845      17209        2517209    Vieirópolis  1462  25126342.0       2.0   
5846       5501        2505501  Vista Serrana  1463  25014226.0       2.0   
5855      17407        2517407         Zabelê  1464  25048368.0       2.0   

      localizacao                                 nome_escola  \
1243          1.0                      ECI EEM JOSE NOMINANDO   
1267          1.0               ECI EEM AGENOR MENDES PEDROSA   
1284          1.0                 EEEFM PADRE HILDON BANDEIRA   
1286          1.0          EEEF DE DEMONSTRACAO ALAGOA GRANDE   
1291          1.0  ECIT OSWALDO TRIGUEIRO DE ALBUQUERQUE MELO   
...           ...                                         ...   
5828          1.0              EEEM PROF ODILON DE FIGUEIREDO   
58

In [19]:
#Q4:Junte ao objeto df, os dados da tabela fato_rendimento_escolar, filtrando o período de 2015 a 2021.

df2=pd.merge(df,rend_escolar, on='cod_escola', how='inner')
df2=df2.query('Anos!=2014').replace(np.nan,0)
df2

UF  Nome_UF  cod_Região_Intermediária Nome_Região_Intermediária  \
1     25  Paraíba                      2503                     Patos   
2     25  Paraíba                      2503                     Patos   
3     25  Paraíba                      2503                     Patos   
4     25  Paraíba                      2503                     Patos   
5     25  Paraíba                      2503                     Patos   
...   ..      ...                       ...                       ...   
3314  25  Paraíba                      2502            Campina Grande   
3315  25  Paraíba                      2502            Campina Grande   
3316  25  Paraíba                      2502            Campina Grande   
3317  25  Paraíba                      2502            Campina Grande   
3318  25  Paraíba                      2502            Campina Grande   

      cod_Região_Imediata Nome_Região_Imediata  cod_Mesorregião  \
1                  250009                Patos                1   
2                  250009                Patos                1   
3                  250009                Patos                1   
4                  250009                Patos                1   
5                  250009                Patos                1   
...                   ...                  ...              ...   
3314               250007             Monteiro                2   
3315               250007             Monteiro                2   
3316               250007             Monteiro                2   
3317               250007             Monteiro                2   
3318               250007             Monteiro                2   

      Nome_Mesorregião  cod_Microrregião  Nome_Microrregião  ...  \
1     Sertão Paraibano                 7  Serra do Teixeira  ...   
2     Sertão Paraibano                 7  Serra do Teixeira  ...   
3     Sertão Paraibano                 7  Serra do Teixeira  ...   
4     Sertão Paraibano                 7  Serra do Teixeira  ...   
5     Sertão Paraibano                 7  Serra do Teixeira  ...   
...                ...               ...                ...  ...   
3314         Borborema                10   Cariri Ocidental  ...   
3315         Borborema                10   Cariri Ocidental  ...   
3316         Borborema                10   Cariri Ocidental  ...   
3317         Borborema                10   Cariri Ocidental  ...   
3318         Borborema                10   Cariri Ocidental  ...   

      cod_municipio  Nome_Município    id  cod_escola  cod_rede  localizacao  \
1           2500106     Água Branca  1242  25033204.0       2.0          1.0   
2           2500106     Água Branca  1242  25033204.0       2.0          1.0   
3           2500106     Água Branca  1242  25033204.0       2.0          1.0   
4           2500106     Água Branca  1242  25033204.0       2.0          1.0   
5           2500106     Água Branca  1242  25033204.0       2.0          1.0   
...             ...             ...   ...         ...       ...          ...   
3314        2517407          Zabelê  1464  25048368.0       2.0          1.0   
3315        2517407          Zabelê  1464  25048368.0       2.0          1.0   
3316        2517407          Zabelê  1464  25048368.0       2.0          1.0   
3317        2517407          Zabelê  1464  25048368.0       2.0          1.0   
3318        2517407          Zabelê  1464  25048368.0       2.0          1.0   

                                 nome_escola situacao_funcionamento  Anos  \
1                     ECI EEM JOSE NOMINANDO                    1.0  2015   
2                     ECI EEM JOSE NOMINANDO                    1.0  2016   
3                     ECI EEM JOSE NOMINANDO                    1.0  2017   
4                     ECI EEM JOSE NOMINANDO                    1.0  2018   
5                     ECI EEM JOSE NOMINANDO                    1.0  2019   
...                                      ...                    ...   ...   
3314  E

In [20]:
#Q5:Imprima uma relação com o nome da escola, nome do município e a taxa de abandono das 20 escolas com
#maior nível de abandono escolar em 2021.
q5=df2.query('Anos==2021').\
filter(items=['nome_escola','Nome_Município','Taxa_de_Abandono']).\
sort_values('Taxa_de_Abandono', ascending= False).\
head(20)
q5

nome_escola          Nome_Município  \
1745                EEEFM BENEDITA TARGINO MARANHAO             João Pessoa   
1580          EEEF PROF MARIA GENY DE SOUSA TIMOTEO             João Pessoa   
3097                          EEEFM ARLINDO RAMALHO                 Solânea   
2784                  EEEFM JOAO SILVEIRA GUIMARAES               São Bento   
361                     EEEFM PROF JOAQUIM UMBELINO               Bom Jesus   
656           EEEFM POETA CARLOS DRUMOND DE ANDRADE          Campina Grande   
2227                   ENE D EXPEDITO E DE OLIVEIRA                   Patos   
871             EEEFM MAJOR VENEZIANO VITAL DO REGO          Campina Grande   
1914                  EEEF II E EJA UMBELINA GARCEZ              Mamanguape   
997                EEEFM PROF JOAO DA CUNHA VINAGRE                   Conde   
1596             EEEM PROF ORLANDO CAVALCANTI GOMES             João Pessoa   
332                              EEEFM AMERICO MAIA  Belém do Brejo do Cruz   
1301               EEEM ADALGISA TEODULO DA FONSECA              Itaporanga   
2891  E EST DE CURSO NORMAL EM NIVEL MEDIO SAO JOSE    São José de Piranhas   
1958               EEIEF INDIO PEDRO MAXIMO DE LIMA                Marcação   
1557              EEEF PROFESSOR JOAO JOSE DA COSTA             João Pessoa   
1516                         ECI HORACIO DE ALMEIDA             João Pessoa   
2030         EEEFM ANTONIO MOACIR DANTAS CAVALCANTI                Maturéia   
1396                              ECIT RAUL MACHADO             João Pessoa   
204                              EEIEFM AKAJUTIBIRO         Baía da Traição   

      Taxa_de_Abandono  
1745              39.6  
1580              38.1  
3097              31.4  
2784              29.1  
361               27.6  
656               24.7  
2227              24.6  
871               23.6  
1914              23.1  
997               20.9  
1596              19.8  
332               19.7  
1301              19.6  
2891              18.7  
1958              18.1  
1557              17.7  
1516              17.3  
2030              16.9  
1396              16.8  
204               16.1

In [21]:
#Q6:Imprima uma relação com o nome da escola, nome do município, nome da mesorregião e a taxa de abandono
#das escolas com o maior nível de abandono escolar em 2021 em cada microrregião do estado da Paraíba.
q6=df2.sort_values('Taxa_de_Abandono', ascending=False).\
query('Anos==2021').\
filter(items=['nome_escola','Nome_Município','Nome_Mesorregião','Taxa_de_Abandono','Nome_Microrregião'])
q6.drop_duplicates(subset=['Nome_Microrregião'])



nome_escola          Nome_Município  \
1745           EEEFM BENEDITA TARGINO MARANHAO             João Pessoa   
3097                     EEEFM ARLINDO RAMALHO                 Solânea   
2784             EEEFM JOAO SILVEIRA GUIMARAES               São Bento   
361                EEEFM PROF JOAQUIM UMBELINO               Bom Jesus   
656      EEEFM POETA CARLOS DRUMOND DE ANDRADE          Campina Grande   
2227              ENE D EXPEDITO E DE OLIVEIRA                   Patos   
1914             EEEF II E EJA UMBELINA GARCEZ              Mamanguape   
1301          EEEM ADALGISA TEODULO DA FONSECA              Itaporanga   
2030    EEEFM ANTONIO MOACIR DANTAS CAVALCANTI                Maturéia   
71                ECIT RENATO RIBEIRO COUTINHO                Alhandra   
1869              EEEF PREF JOSE ANTONIO NEVES              Logradouro   
228                ENE PROF PEDRO A DE ALMEIDA              Bananeiras   
1041             EEEFM FLAVIO RIBEIRO COUTINHO  Cruz do Espírito Santo   
2074                    EEEF MIGUEL SANTA CRUZ                Monteiro   
2327                  EEEFM FELIPE TIAGO GOMES                   Picuí   
583         EEEFM MARIA SOLEDADE ASSIS FREITAS           Cajazeirinhas   
1783                          EEEFM DOM ADAUTO           Juarez Távora   
2593                      EEEM FENELON NOBREGA              Salgadinho   
260    EEE FUNDAMENTAL E MEDIO MELQUIADES TEJO     Barra de São Miguel   
2046                   ECI MARIA JOSE DE SOUZA                Montadas   
118             ECI EEM MONSENHOR JOSE PAULINO                   Arara   
2304                         ENE SANTO ANTONIO                  Piancó   
2100  ECIT FRANCISCO DE ALBUQUERQUE MONTENEGRO                  Natuba   

       Nome_Mesorregião  Taxa_de_Abandono           Nome_Microrregião  
1745     Mata Paraibana              39.6                 João Pessoa  
3097  Agreste Paraibano              31.4          Curimataú Oriental  
2784   Sertão Paraibano              29.1             Catolé do Rocha  
361    Sertão Paraibano              27.6                  Cajazeiras  
656   Agreste Paraibano              24.7              Campina Grande  
2227   Sertão Paraibano              24.6                       Patos  
1914     Mata Paraibana              23.1               Litoral Norte  
1301   Sertão Paraibano              19.6                  Itaporanga  
2030   Sertão Paraibano              16.9           Serra do Teixeira  
71       Mata Paraibana              16.0                 Litoral Sul  
1869  Agreste Paraibano              13.1                   Guarabira  
228   Agreste Paraibano              12.8             Brejo Paraibano  
1041     Mata Paraibana              11.9                        Sapé  
2074          Borborema              11.6            Cariri Ocidental  
2327          Borborema              10.1   Seridó Oriental Paraibano  
583    Sertão Paraibano               9.6                       Sousa  
1783  Agreste Paraibano               8.9                   Itabaiana  
2593          Borborema               6.2  Seridó Ocidental Paraibano  
260           Borborema               6.1             Cariri Oriental  
2046  Agreste Paraibano               5.6                   Esperança  
118   Agreste Paraibano               5.4         Curimataú Ocidental  
2304   Sertão Paraibano               4.6                      Piancó  
2100  Agreste Paraibano               2.1                   Umbuzeiro

In [36]:
#Q7:Imprima uma relação com o código da escola, nome da escola, localização e a diferença da taxa de abandono
#das 10 escolas com as maiores diferenças das taxas de abandono entre os anos de 2019 (antes da pandemia)
#e 2021 (no período da pandemia).

ano_19=df2.query('Anos==2019')
ano_21=df2.query('Anos==2021')
td=ano_19.merge(ano_21,on=['cod_escola','nome_escola','localizacao'])
q7=td.filter(items=['cod_escola','nome_escola','localizacao'])
q7['diff_2019|2021']=td['Taxa_de_Abandono_y']-td['Taxa_de_Abandono_x']
q7.sort_values('diff_2019|2021', ascending=False, inplace=True)

q7.head(10)


cod_escola                            nome_escola  localizacao  \
370  25003844.0          EEEFM JOAO SILVEIRA GUIMARAES          1.0   
48   25006029.0            EEEFM PROF JOAQUIM UMBELINO          1.0   
87   25072056.0  EEEFM POETA CARLOS DRUMOND DE ANDRADE          1.0   
298  25021427.0           ENE D EXPEDITO E DE OLIVEIRA          1.0   
410  25060406.0                  EEEFM ARLINDO RAMALHO          1.0   
235  25123696.0        EEEFM BENEDITA TARGINO MARANHAO          1.0   
175  25030566.0       EEEM ADALGISA TEODULO DA FONSECA          1.0   
213  25094629.0  EEEF PROF MARIA GENY DE SOUSA TIMOTEO          1.0   
141  25088572.0          EEEFM FLAVIO RIBEIRO COUTINHO          1.0   
206  25094190.0                      ECIT JOAO GOULART          1.0   

     diff_2019|2021  
370            21.4  
48             19.6  
87             18.6  
298            17.2  
410            16.8  
235            16.1  
175            14.5  
213            14.2  
141            11.9  
206            11.3

In [37]:
#tabela para a questão 8 com todas as escolas 
J=pd.merge(df,rend_escolar, on='cod_escola', how='inner').\
query('Nome_UF=="Paraíba"').\
query('situacao_funcionamento == 1').\
query('Anos!=2014')


In [38]:
#Q8:Qual é o percentual de escolas com abandono escolar acima da média estadual no ano de 2021?
#calculo
#média de abandono das escolas estaduais :
media_estadual=df2.query('Anos==2021').\
filter(items=['Taxa_de_Abandono'])
d=media_estadual.mean()
print(d)

#escolas que atendem o pré-requisito de ter a média maior que a média estadual:
escolas_2021=J.query('Anos==2021').\
filter(items=['Taxa_de_Abandono'])
d2=escolas_2021[escolas_2021.Taxa_de_Abandono > 3.079565 ]
d2
len(d2)
#todas as escolas :
td_escolas=J.query('Anos==2021').\
filter(items=['Taxa_de_Abandono'])
len(td_escolas)
print(len(d2),len(td_escolas))

Taxa_de_Abandono    3.079565
dtype: float64
136 460


In [39]:
### Q8 resposta
#percetual das escolas 460-100% assim como 136-x%
q8=(len(d2)*100)/len(td_escolas)
print('O percentual de escolas com abandono escolar acima da média estadual no ano de 2021 é de',q8 ,'%.')

O percentual de escolas com abandono escolar acima da média estadual no ano de 2021 é de 29.565217391304348 %.


In [40]:
#Q9:Qual é a média da taxa de abandono escolar por ano?

media_2021=df2.query('Anos==2021').filter(items=['Taxa_de_Abandono']).mean()
print('taxa de abandono de 2021',media_2021)
print('-'*60)
media_2020=df2.query('Anos==2020 ').filter(items=['Taxa_de_Abandono']).mean()
print('taxa de abandono de 2020',media_2020)
print('-'*60)
media_2019=df2.query('Anos==2019 ').filter(items=['Taxa_de_Abandono']).mean()
print('taxa de abandono de 2019',media_2019)
print('-'*60)
media_2018=df2.query('Anos==2018 ').filter(items=['Taxa_de_Abandono']).mean()
print('taxa de abandono de 2018',media_2018)
print('-'*60)
media_2017=df2.query('Anos==2017 ').filter(items=['Taxa_de_Abandono']).mean()
print('taxa de abandono de 2017',media_2017)
print('-'*60)
media_2016=df2.query('Anos==2016 ').filter(items=['Taxa_de_Abandono']).mean()
print('taxa de abandono de 2016',media_2016)
print('-'*60)
media_2015=df2.query('Anos==2015 ').filter(items=['Taxa_de_Abandono']).mean()
print('taxa de abandono de 2015',media_2015)
print('-'*60)


taxa de abandono de 2021 Taxa_de_Abandono    3.079565
dtype: float64
------------------------------------------------------------
taxa de abandono de 2020 Taxa_de_Abandono    2.070925
dtype: float64
------------------------------------------------------------
taxa de abandono de 2019 Taxa_de_Abandono    6.956787
dtype: float64
------------------------------------------------------------
taxa de abandono de 2018 Taxa_de_Abandono    10.357275
dtype: float64
------------------------------------------------------------
taxa de abandono de 2017 Taxa_de_Abandono    12.187374
dtype: float64
------------------------------------------------------------
taxa de abandono de 2016 Taxa_de_Abandono    14.707532
dtype: float64
------------------------------------------------------------
taxa de abandono de 2015 Taxa_de_Abandono    13.634483
dtype: float64
------------------------------------------------------------


In [41]:
#Q10: Qual é a média da taxa de abandono escolar por ano e localização (rural, urbana)?
#rural('localizacao==2')
#urbana('localizacao==1')
#escolas rurais 
print('rural 2021',df2.query('localizacao==2 & Anos==2021').filter(items=['Taxa_de_Abandono']).mean())
print('rural 2020',df2.query('localizacao==2 & Anos==2020').filter(items=['Taxa_de_Abandono']).mean())
print('rural 2019',df2.query('localizacao==2 & Anos==2019').filter(items=['Taxa_de_Abandono']).mean())
print('rural 2018',df2.query('localizacao==2 & Anos==2018').filter(items=['Taxa_de_Abandono']).mean())
print('rural 2017',df2.query('localizacao==2 & Anos==2017').filter(items=['Taxa_de_Abandono']).mean())
print('rural 2016',df2.query('localizacao==2 & Anos==2016').filter(items=['Taxa_de_Abandono']).mean())
print('rural 2015',df2.query('localizacao==2 & Anos==2015').filter(items=['Taxa_de_Abandono']).mean())

print('_'*60)
  
#escolas urbanas 
print('urbana 2021',df2.query('localizacao==1 & Anos==2021').filter(items=['Taxa_de_Abandono']).mean())
print('urbana 2020',df2.query('localizacao==1 & Anos==2020').filter(items=['Taxa_de_Abandono']).mean())
print('urbana 2019',df2.query('localizacao==1 & Anos==2019').filter(items=['Taxa_de_Abandono']).mean())
print('urbana 2018',df2.query('localizacao==1 & Anos==2018').filter(items=['Taxa_de_Abandono']).mean())
print('urbana 2017',df2.query('localizacao==1 & Anos==2017').filter(items=['Taxa_de_Abandono']).mean())
print('urbana 2016',df2.query('localizacao==1 & Anos==2016').filter(items=['Taxa_de_Abandono']).mean())
print('urbana 2015',df2.query('localizacao==1 & Anos==2015').filter(items=['Taxa_de_Abandono']).mean())


rural 2021 Taxa_de_Abandono    2.534884
dtype: float64
rural 2020 Taxa_de_Abandono    1.697619
dtype: float64
rural 2019 Taxa_de_Abandono    6.597368
dtype: float64
rural 2018 Taxa_de_Abandono    9.587179
dtype: float64
rural 2017 Taxa_de_Abandono    9.745946
dtype: float64
rural 2016 Taxa_de_Abandono    14.1
dtype: float64
rural 2015 Taxa_de_Abandono    12.417647
dtype: float64
____________________________________________________________
urbana 2021 Taxa_de_Abandono    3.135731
dtype: float64
urbana 2020 Taxa_de_Abandono    2.108981
dtype: float64
urbana 2019 Taxa_de_Abandono    6.990594
dtype: float64
urbana 2018 Taxa_de_Abandono    10.433503
dtype: float64
urbana 2017 Taxa_de_Abandono    12.438997
dtype: float64
urbana 2016 Taxa_de_Abandono    14.768286
dtype: float64
urbana 2015 Taxa_de_Abandono    13.755102
dtype: float64


Q11: Com base nas análises de dados realizadas, o que é possível concluir sobre os efeitos da pandemia no nível
de abandono escolar no ensino médio?


 Com base nos dados sobre a taxa de abandono das escolas estaduais da paraíba, pode-se concluir que a taxa de abandono escolar de modo geral vem diminuindo ao longo dos anos  ,vemos esse fenômeno ao comparar as taxas de 2015 com as de 2021 que são respectivamente de 13.63% e 3.07% ,contudo, essa diminuição não é linear por ter  aumentos nos anos 2016 e 2021, mas ao comparar a taxa dos anos inicial e final dos dados temos uma queda de mais de 10%.
 No âmago pré e pós pandemia houve uma diminuição de quase 50% na taxa de abandono ao comparar 2019 com 6.95% e 2021 com 3.07%,já no período pandêmico de 2020 a 2021 houve um aumento a taxa era de 2,07% e passou a ser de 3,07% .  
 Nas escolas estaduais de localização urbanas da Paraíba temos um  aumento de 2015 para 2016 e  uma diminuição linear de 2016 à 2020 e um aumento em 2021. O mesmo fenômeno acontece nas escolas de localização Rural da Paraíba .Ao comparar a taxa de abandono das escolas urbanas  com as escolas rurais vemos que  em todo o  período da pesquisa os dados demonstram que as escolas urbanas sofrem em média mais abandonos  que as escolas rurais . 
 As mesorregiões que mais sofrem com o abandono escolar são o Agreste paraibano e o Sertão paraibano ,as escolas dessas duas mesorregiões ocupam 15 das 23  escolas com o maior nível de abandono escolar em 2021.
 
 O efeito da Pandemia nas escolas estaduais de ensino médio da Paraíba na  taxa de abandono escolar foi uma oscilação por ter uma queda significativa de 2019 para 2020  e um aumento em 2021  que não superam os valores de antes da pandemia, então pode-se concluir que até o ano de 2021 a pandemia não afetou excessivamente as taxas de abandono mas hove uma queda considerável.

